### PR TITLE
refactor: files structure

### PR DESCRIPTION
**Pull request by Uzlopak**
improved typings
disable package-lock generation
remove npmignore in favor of setting the files in package.json
validate the chosen algorithm at plugin registration time (prevents from server crashing on first request)

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,15 +3,23 @@
...
+    "lint": "standard | snazzy",
+    "test": "npm run test:unit && npm run test:typescript",
+    "test:unit": "tap",
+    "test:typescript": "tsd"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/fastify/fastify-etag.git"
   },
+  "files": [
```
Comment (climba03003) [inline package.json:18]: It is not identical to `.npmignore` and excluded the `LICENSE` file.

**Review by climba03003 — CHANGES_REQUESTED** [review]

**Comment by Uzlopak** [comment]
Readme.md and LIcense are always included by npm publish

**Review by climba03003 — APPROVED** [review]
LGTM, expect changing `blacklist` to `whitelist` approach for `publish files`.
I will wait for one more people before merging.

**Review by Fdawgs — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -3,15 +3,23 @@
...
+    "index.js",
+    "types/index.d.ts",
+    "fnv1a.js"
+  ],
```
Comment (mcollina) [inline package.json:22]: Let's avoid this. It's just a source of bugs.

Comment (mcollina) [reply package.json:22]: Also... include the test folder in the package.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Uzlopak) [reply package.json:22]: I know we had this discussion regarding tests already. I would prefer not to decide which files should be in the npm package. So i would like to remove the files section in package.json.

**Comment by Uzlopak** [comment]
@mcollina ptal

**Review by Eomm — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**